photutils/psf/photometry.py

```diff
@@ -749,8 +749,16 @@ def _calc_fit_metrics(self, data, source_tbl):
                 qfit.append(np.sum(np.abs(residual)) / flux_fit)
 
                 if np.isnan(cen_idx_):
-                    # calculate residual at central pixel
-                    cen_residual = data[ycen, xcen] - model(xcen, ycen)
+                    # calculate residual at central pixel if the central pixel
+                    # is within the bounds of the ``data``, otherwise mask it:
+                    cen_in_data = (
+                        0 <= ycen <= data.shape[0] - 1 and
+                        0 <= xcen <= data.shape[1] - 1
+                    )
+                    if cen_in_data:
+                        cen_residual = data[ycen, xcen] - model(xcen, ycen)
+                    else:
+                        cen_residual = np.ma.masked
```

Comment:
My slight preference would be to return `np.nan` here.  And it would also be nice to add a new flag (see `_define_flags`) for this condition:  '32' for center pixel is outside of the image.